### ISSUE TITLE
[Suggestion] Config options for dimensional spawning blacklist/whitelist

### ISSUE THREAD

**Issue by katubug** [capnkirok/animaniamod#376]
Hi! Since a few people have asked and it's something I've been interested in myself, would it be possible to add a blacklist or whitelist option for Animania spawns in specific dimensions? The ability to blacklist an entire dimension from allowing Animania spawns would be really helpful for packs with dimensions that use the vanilla biome tags but are thematically incompatible with livestock.

**Comment by Tschipp** [comment]
The spawning config has been annoying me for a while. The config is cluttered and hard to read/understand. I’ve been thinking about making a custom solution involving json files or something like that.

**Comment by Tschipp** [comment]
Would something like this be something you want? I want the mod to be very configurable, but I also don't want to clutter the config any more than necessary. It is massive as it is, and it won't get easier to read when addons come out. I think some json configuration system would be a smart move.

**Comment by katubug** [comment]
Personally, I'm very happy with json configs. As long as the documentation has an example for reverse-engineering, I think it'd be fine for most people.

**Comment by katubug** [comment]
If you don't mind, I'm going to mock up a potential config change that organizes things a little differently and explains things more.

**Comment by katubug** [comment]
It's done! Here it is: https://pastebin.com/2S7bxMHY

As I note at the top, any major changes or things I wasn't sure about are surrounded with `&&ampersands&&` so they stick out. I kept the main categories the same because I assumed it would be a pain to move them around.

I mostly just organized things by category or by animal or both. I added a line for each value stating what the default is (I personally love when configs do this). I also added a little extra description for some values to clear up commonly asked questions.

The way I guessed you might do it is to move all the modular spawning stuff (biomes, dimensions etc) into jsons? But for people unfamiliar with json, it might be nice to have just an "overworld only" spawning option, so I put one in there as a suggestion.

Anyway, hopefully that helps a little. You're welcome to disregard it if it doesn't work for you. :)
